sugar_test.go

```diff
@@ -0,0 +1,226 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package zap
+
+import (
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSugarGetSugarFields(t *testing.T) {
+	var (
+		fields []Field
+		err    error
+	)
+
+	_, err = getSugarFields("test")
+	assert.Error(t, err, "Should return error on invalid number of arguments")
+
+	_, err = getSugarFields("test1", 1, "test2")
+	assert.Error(t, err, "Should return error on invalid number of arguments")
+
+	_, err = getSugarFields("test1", 1, "error", errors.New(""))
+	assert.Error(t, err, "Should return error when error passed as value (special case of unknown type)")
+
+	_, err = getSugarFields(1, 1)
+	assert.Error(t, err, "Should return error on non-string field name")
+
+	fields, _ = getSugarFields("test", 1)
+	assert.Len(t, fields, 1, "Should return 1 field")
+
+	fields, _ = getSugarFields("test1", 1, "test2", 2)
+	assert.Len(t, fields, 2, "Should return 2 fields")
+
+	fields, _ = getSugarFields(errors.New("error"), "test1", 1)
+	assert.Len(t, fields, 2, "Should return 2 fields")
+}
+
+func TestSugarLevel(t *testing.T) {
+	assert.Equal(t, DebugLevel, NewSugar(New(NewJSONEncoder(), DebugLevel)).Level())
+	assert.Equal(t, FatalLevel, NewSugar(New(NewJSONEncoder(), FatalLevel)).Level())
+}
+
+func TestSugarSetLevel(t *testing.T) {
+	sugar := NewSugar(New(NewJSONEncoder()))
+	sugar.SetLevel(FatalLevel)
+	assert.Equal(t, FatalLevel, sugar.Level())
+}
+
+func withSugarLogger(t testing.TB, opts []Option, f func(Sugar, *testBuffer)) {
+	sink := &testBuffer{}
+	errSink := &testBuffer{}
+
+	allOpts := make([]Option, 0, 3+len(opts))
+	allOpts = append(allOpts, DebugLevel, Output(sink), ErrorOutput(errSink))
+	allOpts = append(allOpts, opts...)
+	logger := New(newJSONEncoder(NoTime()), allOpts...)
+	sugar := NewSugar(logger)
+
+	f(sugar, sink)
+}
+
+type m9e struct {
+	foo int
+	bar bool
+}
```

Comment:
You need this struct to be json formattable. This will fix the failing TestSugarLogTypes test  - 

`type m9e struct {
	Foo int  `json:"foo"`
	Bar bool `json:"bar"`
}`